### PR TITLE
feat: `unicode-range` node

### PR DESCRIPTION
**Pull request by evilebottnawi**
No issue, it is part of css spec.

Spec: https://drafts.csswg.org/css-fonts-3/#urange-value
MDN: https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range

Please check one:
- [x] New tests created for this change
- [ ] Tests updated for this change

---

Based on https://github.com/prettier/prettier/issues/4063

### CODE REVIEW

**Comment by shellscape** [comment]
Thanks for the PR 🍺  I'll check this out this afternoon.

File: API.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ not documented here are subject to change at any point.*
   * [`parser.string([props])`](#parserstringprops)
   * [`parser.value([props])`](#parservalueprops)
   * [`parser.word([props])`](#parserwordprops)
+  * [`parser.unicoderange([props])`](#parserunicoderangeprops)
```
Comment (shellscape) [inline API.md]: `atword` is probably an offender and should be changed, but please rename this using camelCase to `unicodeRange`

Comment (shellscape) [reply API.md]: also, "u" comes before "v" :) please place that in alphabetic order

File: API.md [hunk 2]
```diff
@@ -214,13 +215,26 @@ not documented here are subject to change at any point.*
...
+  * `props (object)`: The new node's properties.
+  
 ## Node types
 
 ### `node.type`
 
   A string representation of the node type. It can be one of the following;
   `atword`, `colon`, `comma`, `comment`, `func`, `number`, `operator`,
-  `paren`, `string`, `value`, `word`.
+  `paren`, `string`, `value`, `word`, `unicoderange`.
```
Comment (shellscape) [inline API.md]: "u" before "v" - please place that in alphabetic order

File: API.md [hunk 3]
```diff
@@ -494,7 +508,7 @@ methods are:
 * `container.walkParenthesis`
 * `container.walkStringNodes`
 * `container.walkWords`
-
+* `container.walkUnicodeRanges`
```
Comment (shellscape) [inline API.md]: "u" before "w" - please place that in alphabetic order

File: lib/index.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ const Paren = require('./paren');
 const Str = require('./string');
 const Value = require('./value');
 const Word = require('./word');
+const UnicodeRange = require('./unicode-range');
```
Comment (shellscape) [inline lib/index.js]: "u" before "v" - please place that in alphabetic order

File: lib/index.js [hunk 5]
```diff
@@ -65,4 +66,8 @@ parser.word = function (opts) {
   return new Word(opts);
 };
 
+parser.unicoderange = function (opts) {
```
Comment (shellscape) [inline lib/index.js]: (see previous comment about camelCase)

**Comment by evilebottnawi** [comment]
@shellscape done :+1:

File: API.md [hunk 6]
```diff
@@ -214,13 +215,26 @@ not documented here are subject to change at any point.*
 
   * `props (object)`: The new node's properties.
 
+### `parser.unicoderange([props])`
```
Comment (shellscape) [inline API.md]: `parser.unicodeRange` please

and same with the example on line 223

**Comment by evilebottnawi** [comment]
@shellscape stupid error :smile: done :+1:

File: API.md (hunk 6 above)
Comment (shellscape) [reply API.md]: you missed this line still 😆

**Comment by evilebottnawi** [comment]
@shellscape tired, without sleep around 36 hours :tired_face:

**Comment by shellscape** [comment]
Haha go to bed!! Thanks for the PR. We're all set tp merge.

**Comment by pgrm** [comment]
thx for the effor @evilebottnawi 

@shellscape will there be a 1.3.3 release with this fix soonish? 🙏

**Pull request merged**